pontoon/base/models.py

```diff
@@ -377,6 +357,87 @@ def add_translation(self, string, locale, user):
 
         translation.save()
 
+    def get_translation(self, plural_form=None):
+        """Get fetched translation of a given entity."""
+        translations = self.fetched_translations
+
+        if translations:
+            if plural_form:
+                translations = [t for t in translations if t.plural_form == plural_form]
+
+            translation = sorted(translations, key=lambda k: (-k.approved, k.date))[0]
+
+            return {
+                'fuzzy': translation.fuzzy,
+                'string': translation.string,
+                'approved': translation.approved,
+                'pk': translation.pk
+            }
+
+        else:
+            return {
+                'fuzzy': False,
+                'string': u'',
+                'approved': False,
+                'pk': None
+            }
+
+    @classmethod
```

Comment:
Generally methods like this (that fetch a set of model instances) go on the Manager class, meaning you'll need to make a [custom manager](https://docs.djangoproject.com/en/1.8/topics/db/managers/#custom-managers).